src/APK/Podman.hs

```diff
@@ -0,0 +1,186 @@
+
+module APK.Podman
+  ( Config(..)
+  , HostPaths(..)
+  , PackageName(..)
+  , build
+  , getHostPaths
+  , index
+  , run
+  , update
+  )
+where
+
+import APK.Prelude
+import Control.Monad (unless)
+import Data.Text.Encoding (decodeUtf8, encodeUtf8)
+import System.Exit (ExitCode(ExitSuccess))
+import System.Path ((</>))
+
+import qualified APK.TH                as TH
+import qualified Data.ByteString.Lazy  as LBS
+import qualified Data.List             as List
+import qualified Network.AWS           as AWS
+import qualified Network.AWS.Data.Body as AWS
+import qualified System.Path           as Path
+import qualified System.Path.Directory as Path
+import qualified System.Process.Typed  as Process
+
+newtype ContainerName = ContainerName Text
+  deriving newtype ToText
+
+newtype ImageName = ImageName Text
+  deriving newtype ToText
+
+newtype PackageName = PackageName Text
+  deriving newtype ToText
+
+data Config = Config
+  { arguments     :: [String]
+  , command       :: String
+  , repositoryPath :: Path.RelDir
+  , workPath       :: Path.AbsDir
+  }
+
+data HostPaths = HostPaths
+  { hostAbuildPath             :: Path.AbsDir
+  , hostCachePath              :: Path.AbsDir
+  , hostHomePath               :: Path.AbsDir
+  , hostPackagesPath           :: Path.AbsDir
+  , hostRepositoryPackagesPath :: Path.AbsDir
+  , hostRepositoryPath         :: Path.AbsDir
+  , hostWorkingPath            :: Path.AbsDir
+  }
+
+defaultConfig :: Config
+defaultConfig = Config
+  { arguments      = []
+  , command        = "exit"
+  , repositoryPath = Path.relDir "repository"
+  , workPath       = containerHomePath
+  }
+
+getHostPaths :: IO HostPaths
+getHostPaths
+  =   mkHostPaths
+  <$> Path.getCurrentDirectory
+  <*> Path.getHomeDirectory
+  where
+    mkHostPaths :: Path.AbsDir -> Path.AbsDir -> HostPaths
+    mkHostPaths hostWorkingPath hostHomePath = HostPaths
+      { hostAbuildPath             = hostHomePath </> Path.relDir ".abuild"
+      , hostCachePath              = hostWorkingPath </> Path.relDir "cache"
+      , hostPackagesPath           = hostWorkingPath </> Path.relDir "packages"
+      , hostRepositoryPackagesPath = hostRepositoryPath </> Path.relDir "packages"
+      , ..
+      }
+      where
+        hostRepositoryPath :: Path.AbsDir
+        hostRepositoryPath = hostWorkingPath </> Path.relDir "repository"
+
+containerAbuildPath :: Path.AbsDir
+containerAbuildPath = containerHomePath </> Path.relDir ".abuild"
+
+containerHomePath :: Path.AbsDir
+containerHomePath = Path.absDir "/opt/build"
+
+containerCachePath :: Path.AbsDir
+containerCachePath = Path.absDir "/var/cache"
+
+containerRepositoryPath :: Path.AbsDir
+containerRepositoryPath = containerHomePath </> Path.relDir "repository"
+
+containerPackagesPath :: Path.AbsDir
+containerPackagesPath = containerHomePath </> Path.relDir "packages"
+
+run :: String -> [String] -> IO ()
+run command arguments = runConfig defaultConfig{arguments = arguments, command = command}
```

Comment:
@mbj do you know if it is valid syntax to use something like this:

```haskell
run command arguments = runConfig defaultConfig{..}
```

Reply:
Record wildcards does not work for record updates.